recorder.go

```diff
@@ -349,21 +353,22 @@ func (r *Recorder) Flush() {
 	r.reportInFlight = false
 	if err != nil {
 		// Restore the records that did not get sent correctly
-		r.counters.droppedSpans += int64(r.buffer.addSpans(rawSpans))
-		r.counters.droppedSpans += droppedPending
+		atomic.AddInt64(&r.counters.droppedSpans, int64(r.buffer.addSpans(rawSpans))+droppedPending)
 		r.lock.Unlock()
 		return
 	}
 
 	// Reset the buffers
 	r.reportOldest = now
 	r.reportYoungest = now
-	// TODO: this ends up discarding counts coming in during the RPC
-	r.counters = counterSet{}
```

Comment:
I'm leaving a TODO in my local copy to remove the atomics. Although we
could probably rewrite this to not take a lock (using a channel?) and then
the atomics would be handy.
ᐧ

On Wed, Jul 6, 2016 at 1:53 PM, spoons <notifications@github.com> wrote:

> In recorder.go
> <https://github.com/lightstep/lightstep-tracer-go/pull/20#discussion_r69809378>
> :
>
> >  		r.lock.Unlock()
> >  		return
> >  	}
> >
> >  	// Reset the buffers
> >  	r.reportOldest = now
> >  	r.reportYoungest = now
> > -	// TODO: this ends up discarding counts coming in during the RPC
> > -	r.counters = counterSet{}
>
> This is definitely a bug, so great. I'm not convinced that the atomic int
> is needed (isn't it always read/written under the lock?) but happy to hear
> more.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/lightstep/lightstep-tracer-go/pull/20/files/0acc78e7b07da6df26a32ffb9b4f4b45ce05b009#r69809378>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADdiiWOdHXbAFFe4YpZvl3hFv7nVPKaHks5qTBW9gaJpZM4JFn3C>
> .
>